plugins/action/action.h

```diff
@@ -24,29 +38,164 @@ class Action
         UNKNOWN
     };
 
+    /**
+     * Returns a human-readable English string for an Action::Result.
+     */
     static const char *result_str(Result);
 
-    typedef std::function<void(Result)> result_callback_t;
-
+    /**
+     * Arms the drone (synchronous).
+     *
+     * Note that arming a drone normally means that the motors will spin at idle.
+     * Therefore, before arming take all safety precautions and stand clear of the drone.
+     *
+     * @return result of request
+     */
     Result arm() const;
+
+    /**
+     * Disarms the drone (synchronous).
+     *
+     * This will disarm a drone if it considers itself landed. If flying in-air, the drone should
+     * reject the disarm command. Disarming means that all motors will stop.
+     *
+     * @return result of request
+     */
     Result disarm() const;
+
+    /**
+     * Kills the drone (synchronous).
+     *
+     * This will disarm a drone no matter if still landed or still in-air.
+     * Note that the drone will fall out of the sky if you use this command while flying.
+     *
+     * @return result of request
+     */
     Result kill() const;
+
+    /**
+     * Commands the drone to take off and hover at a set takeoff altitude (synchronous).
```

Comment:
What value is set for the takeoff altitude? (bit confusing, because the API does not allow you to set it :-) 

I think you probably mean: `Commands the drone to take off and hover at a the vehicle's default takeoff altitude (synchronous).`
... as an aside, why not supply a takeoff altitude with preset default?

For consistency, all actions should have the same "voice" - this one is different:
* Arms the drone (synchronous).
* Kills the drone (synchronous).
* Commands the ...

Reply:
What if the vehicle is not armed when this is called?

Reply:
> Link to set_takeoff_altitude() - so it is clear *how* the set takeoff altitude  is set (you could mean some "vehicle default value")

done

> ... as an aside, why not supply the takeoff altitude in this call too?

Possible but then we'd have to remove the set/get calls.

> For consistency, where possible, actions should have the same "voice" - this one is different:

I agree but I didn't know better.

Reply:
> What if the vehicle is not armed when this is called? Generally are there any preconditions for any of these functions to be called?

Good question. It will probably say `COMMAND_DENIED`. I'll add a note.